pkg/stanza/operator/parser/syslog/syslog.go

```diff
@@ -77,6 +78,8 @@ func (c Config) Build(logger *zap.SugaredLogger) (operator.Operator, error) {
 		return nil, err
 	}
 
+	proto := strings.ToLower(c.Protocol)
```

Comment:
Nice improvement. Should we also use this value in all the other cases?

Reply:
yes, I will update other ocurences

Reply:
I agree this is a big improvement for `syslog` receiver. Previously, running the receiver with e.g. `protocol: RFC5424` would not return any error, but the parsing of logs was not working. Now this works, and for all incorrect protocol names the receiver with fail to start.

I just wish these kind of big improvements were reflected in the changelog with a bit more specificity. Just `add integration tests and fix related bugs` is way to vague to my taste. Perhaps we could have more than one changelog item for this PR? Or add some notes describing more details?

Reply:
Agree, and thanks for the improvement. Arguably this could be a separate PR, but at least we should have another changelog entry in this one.

Reply:
I will create separate PR

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/27581

Reply:
Thanks. Approved and merged the other PR